python/lsst/obs/base/cameraMapper.py

```diff
@@ -1059,6 +1074,156 @@ def getRegistry(self):
+        "scaling" entries. Defaults will be provided for any missing
+        elements under "compression" and "scaling".
+
+        The allowed entries under "compression" are:
+
+        * scheme (string): compression algorithm to use
+        * rows (int): number of rows per tile (0 = entire image)
+        * quantizeLevel (float): cfitsio quantization level
+
+        The allowed entries under "scaling" are:
+
+        * scheme (string): scaling algorithm to use
+        * bitpix (int): bits per pixel (0,8,16,32,64,-32,-64)
+        * fuzz (bool): fuzz the values when quantising floating-point values?
+        * seed (long): seed for random number generator when fuzzing
+        * maskPlanes (list of string): mask planes to ignore when doing statistics
+        * quantizeLevel: divisor of the standard deviation for STDEV_* scaling
+        * quantizePad: number of stdev to allow on the low side (for STDEV_POSITIVE/NEGATIVE)
+        * bscale: manually specified BSCALE (for MANUAL scaling)
+        * bzero: manually specified BSCALE (for MANUAL scaling)
+
+        A very simple example YAML recipe:
+
+            default:
+              image: &default
+                compression:
+                  scheme: GZIP_SHUFFLE
+              mask: *default
+              variance: *default
+
+        This method raises a `RuntimeError` if the YAML contents don't
+        match the expected schema.
+        """
+
+        defaults = os.path.join(getPackageDir("obs_base"), "policy", "writeRecipes.yaml")
+        with open(defaults) as ff:
+            recipes = yaml.load(ff)
+        overrideFile = os.path.join(self.getPackageDir(), "policy", "writeRecipes.yaml")
+        if os.path.exists(overrideFile):
+            with open(overrideFile) as ff:
+                overrides = yaml.load(ff)
+
+            def applyOverride(original, new):
+                for key, value in new.items():
+                    if isinstance(original, collections.Mapping):
+                        if isinstance(value, collections.Mapping):
+                            original[key] = applyOverride(original.get(key, {}), value)
+                        else:
+                            original[key] = new[key]
+                    else:
+                        original = {key: new[key]}
+                return original
+
+            applyOverride(recipes, overrides)
+
+        # Schemas define what should be there, and the default values (and by the default
+        # value, the expected type).
+        compressionSchema = {
+            "scheme": "NONE",
+            "rows": 1,
+            "quantizeLevel": 0.0,
+        }
+        scalingSchema = {
+            "scheme": "NONE",
+            "bitpix": 0,
+            "maskPlanes": ["NO_DATA"],
+            "seed": 0,
+            "quantizeLevel": 4.0,
+            "quantizePad": 5.0,
+            "fuzz": True,
+            "bscale": 1.0,
+            "bzero": 0.0,
+        }
```

Comment:
I wanted to add cerberus (http://docs.python-cerberus.org/en/stable/) to do this kind of validation, for policy files in general (but it could be applied more selectively, like just here) (K-T shot it down because he said users didn't want this level of verification on policy files in general). You should take a look and see if you'd rather use it than have your own validation code here.